avionics/common/include/tasks/main_tasks.hpp

```diff
@@ -30,20 +32,25 @@ class ReadEvalLog {
             state_machine.abort();
         }
 
-        StateId state = state_machine.getState();
+        state_ = state_machine.getState();
+        if(state_ != oldState_) {
```

Comment:
Probably best to throw this in the `update()` method of [state machine](https://github.com/UBC-Rocket/FLARE/blob/dev/avionics/common/include/state_machine.hpp)

Note - I mulled over this and intend to remove the `onEntry()` calls in `arm()`, `disarm()`, and `abort()`, since it makes reasoning about state behaviour more difficult from within the state. But this also means we won't get state packets sent in response of arming - which is in theory fine, because there's acknowledgements sent on the XBee protocol layer (not sure if anything is done with them on FLAREGUN), and also state is sent at regular intervals and on the ground it's not that important. Possibly a bit annoying for testing purposes. 

If not we can also potentially add a command acknowledge packet, which could provide more detail (e.g. whether the command was accepted) - but that's not important now